### PR TITLE
Target host fix

### PR DESCRIPTION
**Pull request by at88mph**
Use target API URL when obtaining AuthorizationToken.

### CODE REVIEW

File: src/main/java/net/canfar/storage/web/resources/GroupNameServerResource.java [hunk 1]
```diff
@@ -120,13 +122,19 @@ public void write(final JSONWriter writer) throws JSONException {
...
                 }
             }
         };
     }
 
-    final List<String> queryGroupNames(final Subject voSpaceUser) throws PrivilegedActionException {
-        return Subject.doAs(voSpaceUser, (PrivilegedExceptionAction<List<String>>) () -> {
+    final List<String> queryGroupNames() throws Exception {
+        final RegistryClient registryClient = getRegistryClient();
+        final URL gmsServiceURL = registryClient.getServiceURL(storageConfiguration.getGMSServiceURI(), Standards.GMS_SEARCH_10, AuthMethod.TOKEN);
```
Comment (SharonGoliath) [inline src/main/java/net/canfar/storage/web/resources/GroupNameServerResource.java:135]: This a very naive question - is there only ever going to be one `RegistryClient`, and one `gmsServiceURL`? If not, where is multiplicity going to be handled?

File: src/main/java/net/canfar/storage/web/resources/StorageItemServerResource.java [hunk 2]
```diff
@@ -473,6 +476,15 @@ <T> T executeSecurely(final Subject subject, final PrivilegedExceptionAction<T>
         }
     }
 
+    String getDisplayName() throws Exception {
+        final IdentityManager identityManager = AuthenticationUtil.getIdentityManager();
+        return identityManager.toDisplayString(getVOSpaceCallingSubject());
+    }
+
+    Subject getVOSpaceCallingSubject() throws Exception {
```
Comment (SharonGoliath) [inline src/main/java/net/canfar/storage/web/resources/StorageItemServerResource.java:484]: Are you on-purpose over-riding the method `getCallingSubject`, without doing language-supported over-riding? If so, perhaps add a comment to say why this is necessary?

**Review by SharonGoliath — APPROVED** [review]

File: src/main/java/net/canfar/storage/web/resources/GroupNameServerResource.java (hunk 1 above)
Comment (at88mph) [reply src/main/java/net/canfar/storage/web/resources/GroupNameServerResource.java:135]: Correct, only one for now.  When we implement multiple GMSes, we will need to ask the Node where to go for its Group information, but we don't have an API for that yet.

Comment (at88mph) [reply src/main/java/net/canfar/storage/web/resources/GroupNameServerResource.java:135]: Comment added to other method.  Thanks!

**Pull request merged**